### PR TITLE
[Publisher] Remove duplicate state codes in agency name within the agency dropdown menu

### LINKED ISSUE #1038: [Publisher] Fix duplicate state abbreviation in agency name within the agency dropdown menu
The FE currently appends the state abbreviation to the agency's name in the agency dropdown menu in this format: `Agency Name (AL)`. 

Since we can't currently have two agencies with the same name, there are some states that have the state abbreviation within their agency name to differentiate similar agency names in different cities. This results in their name being displayed in the agency dropdown menu as: `Agency Name (AL) (AL)`.  The solution would be to add logic to deduplicate the abbreviations if one is already included in the name.

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Removes duplicate state codes in agency's name within the agency dropdown menu. The current logic appends the state code to each agency in the dropdown menu - so if an agency's name is "Department of X (CA)" it will render as "Department of X (CA) (CA)" in the menu. This adjustment checks to see if the state code (in parenthesis) is already included in the agency's name - and if so, it will not append a state code.

Demo (the agency has the state code directly in its name):

https://github.com/Recidiviz/justice-counts/assets/59492998/b0b329a6-38ca-459c-b5e5-7068322d9b9a

## Related issues

Closes #1038

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -89,19 +89,28 @@ const Menu: React.FC = () => {
...
-            handleCloseMobileMenu();
-          },
-          highlight: agency.id === currentAgency?.id,
-        }))
+        .map((agency) => {
+          const stateCodeDisplayName = agency.state_code
+            .split("_")[1]
+            .toUpperCase();
+          const isStateCodeInAgencyName =
+            agency.name.includes(stateCodeDisplayName);
```
Comment (mxosman) [inline publisher/src/components/Menu/Menu.tsx:97]: Created a variable for the state code display name `agency.state_code.split("_")[1].toUpperCase()` and a boolean to check if the state code display name is included within the agency name.

No changes below aside from using these two variables.

File: publisher/src/components/Menu/Menu.tsx [hunk 2]
```diff
@@ -89,19 +89,28 @@ const Menu: React.FC = () => {
...
-            handleCloseMobileMenu();
-          },
-          highlight: agency.id === currentAgency?.id,
-        }))
+        .map((agency) => {
+          const stateCodeDisplayName = agency.state_code
+            ?.split("_")[1]
+            .toUpperCase();
+          const isStateCodeInAgencyName =
+            agency.name.includes(stateCodeDisplayName);
```
Comment (mxosman) [inline publisher/src/components/Menu/Menu.tsx:97]: Created a variable for the state code display name agency.state_code.split("_")[1].toUpperCase() and a boolean to check if the state code display name is included within the agency name.

No changes below aside from using these two variables (adding another conditional check and using the display name var).

**Review by nichelle-hall — APPROVED** [review]

**Review by morden35 — APPROVED** [review]
nice!

**Comment by mxosman** [comment]
Thank you both so much!

**Pull request merged**